src/Apps/Order/Components/BankAccountPicker.tsx

```diff
@@ -45,7 +50,23 @@ export const BankAccountPicker: FC<Props> = props => {
     clientSecret,
   } = props
 
-  const bankAccountsArray = extractNodes(bankAccounts)
+  const bankAccountsArray: BankAccountRecord[] = extractNodes(bankAccounts)
+
+  if (
+    order.paymentMethodDetails &&
+    order.paymentMethodDetails.internalID &&
+    order.paymentMethodDetails.last4
+  ) {
+    // if account on order is not saved on user's profile
+    const isOrderBankSaved = bankAccountsArray.find(
+      bank => bank.last4 === order.paymentMethodDetails?.last4
+    )
+
+    if (!isOrderBankSaved) {
+      // populate banks array with the account on order
+      bankAccountsArray.unshift(order.paymentMethodDetails as BankAccountRecord)
+    }
+  }
```

Comment:
Responding to https://github.com/artsy/force/pull/10928#discussion_r963843402 which is not appearing in the diff, my understanding of what is going on here:

- for the first `if` to enter this code,  `order.paymentMethodDetails` must have an `internalID` and a `last4` - is this because `last4` is unique to bank accounts, so we avoid checking on other payment methods? If so I would make that assumption more explicit, maybe by checking payment method directly.
- `isOrderBankSaved` on L60 makes sense, but I think testing `internalID` makes more sense if it is available on both objects.

Question on L67: Is there any chance that mutating this object could cause strange behavior since `bankAccountsArray` is sourced from relay? I'm thinking the answer is no, since even if the code ran multiple times we would be working with a new array (from `extractNodes`) each time. Probably nothing.

Reply:
@erikdstock, Rachel's comment has a point. I initially made the fields in BankAccountRecord interface nullable (`internalID: string | null...`), therefore I was checking the presence of the fields in `paymentMethodDetails`, but ut's unnecessary now. will remove.

Reply:
what's going on here is that: 

L53: we extract the saved bank accounts on user's profile into `bankAccountsArray`

L55: we check if order has a bankAccount set on it as the paymentMethod (it can only be bank account at this point as you recognised in L201) 

L61: we are trying to find if the bankAccount on order is a saved bank account on user's profile 

L65: if the bank account on order is not saved in users's profile, we are placing it in the beginning of `bankAccountsArray`, so that when user goes to the Review step, clicks on 'Change payment method' and comes back to the payment page, we are able to display the unsaved bank account that they initially set on Order.